setuptools/package_index.py

```diff
@@ -588,24 +588,27 @@ def find(req, env=None):
                     continue
 
                 if dist in req and (dist.precedence<=SOURCE_DIST or not source):
-                    return dist
+                    mylocation = self.download(dist.location, tmpdir)
+                    if os.path.exists(mylocation):
+                        return dist, mylocation
+            return None, None
```

Comment:
sure, I can clean this up, 
 thought this would be ok, since  `process_index` inner  `scan(link)` function already does the same ;-)